hermes-frontend/src/main/java/pl/allegro/tech/hermes/frontend/publishing/handlers/MessageReadHandler.java

```diff
@@ -141,12 +153,25 @@ private void messageRead(HttpServerExchange exchange, byte[] messageContent, Att
             attachment.removeTimeout();
             messageErrorProcessor.sendAndLog(exchange, attachment.getTopic(),
                     attachment.getMessageId(), error(e.getMessage(), VALIDATION_ERROR));
+        } catch (FixedThroughputLimiter.QuotaViolationException e) {
```

Comment:
Why `FixedThroughputLimiter.*` and not`ThroughputLimiter.*`?
I would suggest refactoring `QuotaViolation` exceptions making them proper top level classes.